common/types/activation.go

```diff
@@ -525,3 +399,153 @@ type EpochActiveSet struct {
 }
 
 var MaxEpochActiveSetSize = scale.MustGetMaxElements[EpochActiveSet]("Set")
+
+func (p *Post) ToWireV1() *wire.PostV1 {
+	if p == nil {
+		return nil
+	}
+	return &wire.PostV1{
+		Nonce:   p.Nonce,
+		Indices: p.Indices,
+		Pow:     p.Pow,
+	}
+}
+
+func (n *NIPost) ToWireV1() *wire.NIPostV1 {
+	if n == nil {
+		return nil
+	}
+
+	return &wire.NIPostV1{
+		Membership: *n.Membership.ToWireV1(),
+		Post:       n.Post.ToWireV1(),
+		PostMetadata: &wire.PostMetadataV1{
+			Challenge:     n.PostMetadata.Challenge,
+			LabelsPerUnit: n.PostMetadata.LabelsPerUnit,
+		},
+	}
+}
+
+func (p *MerkleProof) ToWireV1() *wire.MerkleProofV1 {
+	if p == nil {
+		return nil
+	}
+	nodes := make([]Hash32, 0, len(p.Nodes))
+	for _, node := range p.Nodes {
+		nodes = append(nodes, Hash32(node))
+	}
+	return &wire.MerkleProofV1{
+		Nodes:     nodes,
+		LeafIndex: p.LeafIndex,
+	}
+}
+
+func (c *NIPostChallenge) ToWireV1() *wire.NIPostChallengeV1 {
+	return &wire.NIPostChallengeV1{
+		PublishEpoch:   c.PublishEpoch.Uint32(),
+		Sequence:       c.Sequence,
+		PrevATXID:      primitive.Hash32(c.PrevATXID),
+		PositioningATX: primitive.Hash32(c.PositioningATX),
+		CommitmentATX:  (*primitive.Hash32)(c.CommitmentATX),
+		InitialPost:    c.InitialPost.ToWireV1(),
+	}
+}
+
+func (a *ActivationTx) ToWireV1() *wire.ActivationTxV1 {
+	return &wire.ActivationTxV1{
+		InnerActivationTxV1: wire.InnerActivationTxV1{
+			NIPostChallengeV1: *a.NIPostChallenge.ToWireV1(),
+			Coinbase:          wire.Address(a.Coinbase),
+			NumUnits:          a.NumUnits,
+			NIPost:            a.NIPost.ToWireV1(),
+			NodeID:            (*primitive.Hash32)(a.NodeID),
+			VRFNonce:          (*wire.VRFPostIndex)(a.VRFNonce),
+		},
+		SmesherID: Hash32(a.SmesherID),
+		Signature: a.Signature,
+	}
+}
+
+// Decode ActivationTx from bytes.
+// In future it should decide which version of ActivationTx to decode based on the publish epoch.
+func AcivationTxFromBytes(data []byte) (*ActivationTx, error) {
+	var wireAtx wire.ActivationTxV1
+	err := codec.Decode(data, &wireAtx)
+	if err != nil {
+		return nil, fmt.Errorf("decoding ATX: %w", err)
+	}
+
+	return ActivationTxFromWireV1(&wireAtx), nil
+}
+
+func ActivationTxFromWireV1(atx *wire.ActivationTxV1) *ActivationTx {
```

Comment:
I also need it in the SQL code to deserialize the blob. Hopefully, we will remove decoding the blob soon and then it can be moved to the ATX handler code.

Reply:
I want to keep the change much smaller, a lot of the code here is only needed to be removed again in the next PR.

Reply:
Some code (especially decoding the wire type to types.ActivationTx needs to exist now and will be removed when types.ActivationTx is simplified to not contain things like Post).